### PR TITLE
Sometimes when testing a container, a command might not return 0

### PR DESCRIPTION
**Pull request by spolti**
it does not means that the command fail, with that said, the check will
fail unless you modify te test appending "|| true" at the end to force
it to return 0. A good example is if you want to test if a rpm package
is istalled or not in the target container.

If the package exists, there will be no exception, but if the package
does not exists (and this is the intention of the test) it will thrown a
exception and will hide the root cause.

E.g.

```bash
  Scenario: Ensure the openjdk8 packages are not installed on container.
    When container is ready
    Then run sh -c '/usr/bin/rpm -q java-1.8.0-openjdk-devel' in
container and check its output contains package java-1.8.0-openjdk-devel
is not installed
```

In this example, the package is not installed and the test fails unless
|| true is appended at the end.

With this change we will have a better output about why the test has
failed:

```
      Exception: ("Phrase 'package java-1.8.0-openjdk-devel is not
installed' was not found in the output of running the 'sh -c
'/usr/bin/rpm -q java-1.8.0-openjdk-devel'' command", b'package
java-1.8.0-openjdk-devel is not installed\n', ExecException("Command sh
-c '/usr/bin/rpm -q java-1.8.0-openjdk-devel' failed to execute, return
code: 1"))

```

### CODE REVIEW

File: steps/container_steps.py [hunk 1]
```diff
@@ -22,11 +22,13 @@ def log_matches_regex(context, regex, timeout=TIMEOUT):
     if not run_log_matches_regex(context, regex, timeout):
         raise Exception("Regex '%s' did not match the logs" % regex)
 
+
```
Comment (luck3y) [inline steps/container_steps.py:25]: Whitespace changes are generally better off done as a separate commit IMO.

File: steps/container_steps.py [hunk 2]
```diff
@@ -315,8 +321,10 @@ def run_command_expect_message(context, cmd, output_phrase, timeout=80):
                     return True
             except ExecException as e:
                 last_output = e.output
+                exception_cause = e
                 time.sleep(1)
-    raise Exception("Phrase '%s' was not found in the output of running the '%s' command" % (output_phrase, cmd), last_output)
+    raise Exception("Phrase '%s' was not found in the output of running the '%s' command" % (output_phrase, cmd),
```
Comment (luck3y) [inline steps/container_steps.py:329]: I don't think I quite understand the intent here (I get that you're using | true. What will happen to tests using this that are expecting to get the exception?

File: steps/container_steps.py (hunk 1 above)
Comment (spolti) [reply steps/container_steps.py:25]: It is just PEP convention changes made by IDE when formatting it, I can just drop it but, IMHO, doing small formatting changes while fixing other stuff is helpful. your call.

File: steps/container_steps.py (hunk 2 above)
Comment (spolti) [reply steps/container_steps.py:329]: this code change will not change the function's behavior, instead, just append more information on the exception printed to the user instead just tell that the expected text does not match the result.

>> What will happen to tests using this that are expecting to get the exception?

 I could be wrong, but, if it is possible, should behave the same way, but, if you want to check for e ExecException then you'll need to change the except logic to catch that.

Comment (luck3y) [reply steps/container_steps.py:329]: Well, the only reason I ask is if the behaviour changes and we have tests using it, they'll all need to be fixed, so it could be a better idea to add a new test method instead. If it won't break anything that exists already, then no need though.

Comment (spolti) [reply steps/container_steps.py:329]: we have some tests that use it and it seems to be working ok. I'll give it another try and run the tests that use it individually.
On the other hand, do you mind pointing me to some tests that you are afraid to be affected so I can test it as well here just to be sure it will not break anything?

Comment (luck3y) [reply steps/container_steps.py:329]: There are some tests for in the jboss-eap-7-openshift-image and jboss-eap-modules repos, after a brief look they appear to all expect to return something sensible though, so not sure if this change would have an impact if say the command expected was no longer there or something.

Comment (spolti) [reply steps/container_steps.py:329]: this will not change the output or method return, is there a way for you to give it a try with my v1 branch?

**Comment by spolti** [comment]
@goldmann do you mind taking a look? :)

File: steps/container_steps.py (hunk 2 above)
Comment (rnc) [reply steps/container_steps.py:329]: @spolti Can this be merged?
